### PR TITLE
Gateway API: fix handling of various invalid route scenarios

### PR DESCRIPTION
**Pull request by skriss**
Updates the handling of various invalid HTTPRoute/TLSRoute
scenarios to match the Gateway API spec and conformance
tests.

### CODE REVIEW

**Comment by skriss** [comment]
Unfortunately there was one last bug in the v0.5.0-rc2 conformance tests, so this PR still won't pass conformance, but it should pass the nightly `main` conformance run since the bug has been fixed post-rc2 release.

File: changelogs/unreleased/4614-skriss-minor.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Gateway API: update handling of various invalid HTTPRoute/TLSRoute scenarios
+
+Updates the handling of various invalid HTTPRoute/TLSRoute scenarios to be conformant with the Gateway API spec, including:
+
+- Use a 500 response instead of a 404 when a route's backends are invalid
+- The `Accepted` condition on a route only describes whether the route attached successfully to its parent, not whether it has any other errors
+- Use the upstream reasons `InvalidKind` and `BackendNotFound` when a backend is not a Service or not found, respectivelt
```
Comment (youngnick) [inline changelogs/unreleased/4614-skriss-minor.md:7]:
```suggestion
- Use the upstream reasons `InvalidKind` and `BackendNotFound` when a backend is not a Service or not found, respectively
```

**Review by youngnick — APPROVED** [review]
LGTM with one tiny spelling nit.

File: changelogs/unreleased/4614-skriss-minor.md (hunk 1 above)
Comment (skriss) [reply changelogs/unreleased/4614-skriss-minor.md]: doh!

**Pull request merged**